server/channels/api4/cloud.go

```diff
@@ -65,10 +65,15 @@ func (api *API) InitCloud() {
 
 func ensureCloudInterface(c *Context, where string) bool {
 	cloud := c.App.Cloud()
+	disabled := c.App.Config().CloudSettings.Disable != nil && *c.App.Config().CloudSettings.Disable
 	if cloud == nil {
 		c.Err = model.NewAppError(where, "api.server.cws.needs_enterprise_edition", nil, "", http.StatusBadRequest)
 		return false
 	}
+	if disabled {
+		c.Err = model.NewAppError(where, "api.server.cws.disabled", nil, "", http.StatusBadRequest)
```

Comment:
I am 0/5 on changing this, but is `http.StatusBadRequest` what we want to return? Just curious if we had a reason for this over other returns.

Reply:
`BadRequest` might not be the right one. Per https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/400 it's a

> response status code indicates that the server cannot or will not process the request due to something that is perceived to be a client error (for example, malformed request syntax, invalid request message framing, or deceptive request routing).


Maybe 422 Unprocessable Content is better here as it is about semantic errors and 400 about syntactic ones.

> The HyperText Transfer Protocol (HTTP) 422 Unprocessable Content response status code indicates that the server understands the content type of the request entity, and the syntax of the request entity is correct, but it was unable to process the contained instructions.

https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/422